corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -153,3 +156,40 @@ def get_user_stubs(user_ids):
         .user_ids(user_ids)
         .show_inactive()
         .values('_id', 'username', 'first_name', 'last_name', 'doc_type', 'is_active'))
+
+
+def get_form_counts_by_user_xmlns(domain, startdate, enddate, user_ids=None,
+                                  xmlnss=None, by_submission_time=True):
+
+    date_filter_fn = submitted_filter if by_submission_time else completed_filter
+    query = (
+        FormES()
+        .domain(domain)
+        .filter(date_filter_fn(gte=startdate, lt=enddate))
+        .aggregation(
+            TermsAggregation('user_id', 'form.meta.userID').aggregation(
+                TermsAggregation('app_id', 'app_id').aggregation(
+                    TermsAggregation('xmlns', 'xmlns')
+                )
+            )
+        )
+        .size(0)
+    )
+
+    if user_ids:
+        query = query.user_id(user_ids)
+
+    if xmlnss:
+        query = query.xmlns(xmlnss)
+
+    counts = defaultdict(lambda: 0)
+    user_buckets = query.run().aggregations.user_id.buckets_list
+    for user_bucket in user_buckets:
+        app_buckets = user_bucket.app_id.buckets_list
+        for app_bucket in app_buckets:
+            xmlns_buckets = app_bucket.xmlns.buckets_list
+            for xmlns_bucket in xmlns_buckets:
+                key = (user_bucket.key, app_bucket.key, xmlns_bucket.key)
+                counts[key] = xmlns_bucket.doc_count
+
+    return counts
```

Comment:
:+1: :+1: 
I absolutely :cupid: an isolated, tested function wrapping this sort of thing.